### PR TITLE
MRC file format bug and compatibility fixes.

### PR DESCRIPTION
**Pull request by blowekamp**
Addresses issues with the system stamp, 0 -spacing, and remove dead code creating warnings.

This PR is also based on the release branch so that it can also be back patched.

### CODE REVIEW

**Comment by blowekamp** [comment]
The MRC input image file didn't get magically converted to a hash file. It will need to be manually updated.

**Review by zivy — APPROVED** [review]
LGTM

**Review by dzenanz — APPROVED** [review]
That file needs to be replaced by the hash (and uploaded data.kitware.com), otherwise looks fine.

File: Modules/IO/MRC/src/itkMRCImageIO.cxx [hunk 1]
```diff
@@ -200,20 +200,10 @@ MRCImageIO::ReadImageInformation()
...
-  else
-  {
-    m_Spacing[0] = header.xlen / static_cast<float>(header.mx);
-    m_Spacing[1] = header.ylen / static_cast<float>(header.my);
-    m_Spacing[2] = header.zlen / static_cast<float>(header.mz);
-  }
+  // if the spacing was not set in the header then 1.0 is the default
+  m_Spacing[0] = header.xlen != 0.0 ? header.xlen / static_cast<float>(header.mx) : 1.0;
+  m_Spacing[1] = header.ylen != 0.0f ? header.ylen / static_cast<float>(header.my) : 1.0;
+  m_Spacing[2] = header.zlen != 0.0f ? header.zlen / static_cast<float>(header.mz) : 1.0;
```
Comment (N-Dekker) [inline Modules/IO/MRC/src/itkMRCImageIO.cxx:206]: The first `0.0` (in `header.xlen != 0.0`) does not have an `f` postfix, while the others have (`0.0f`). Please double-check if that's intended. Also, the `1.0` constants might have an `f` postfix... anyway, my 2 cent!

File: Modules/IO/MRC/src/itkMRCHeaderObject.cxx [hunk 2]
```diff
@@ -110,55 +110,6 @@ MRCHeaderObject::SetHeader(const Header * buffer)
...
-  {
-    extendedHeaderBytes += 4;
-  }
-  if (this->m_Header.nreal & 1024)
-  {
-    extendedHeaderBytes += 2;
-  }
-  // TODO: all the above to set extendedHeaderBytes, yet extendedHeaderBytes is unused!
-  itkWarningMacro(<< "extendedHeaderBytes is: " << extendedHeaderBytes
-                  << "If you see this log please contact https://github.com/InsightSoftwareConsortium/ITK/pull/2778");
```
Comment (N-Dekker) [inline Modules/IO/MRC/src/itkMRCHeaderObject.cxx:160]: Should we now contact #2778 ? @seanm ? Just curious!

**Comment by dzenanz** [comment]
I uploaded the test file here: https://data.kitware.com/#folder/64413821cffd1a074ef04595

Here is the hash file to be put into the repository:
[tilt_uint8.mrc.sha512.zip](https://github.com/InsightSoftwareConsortium/ITK/files/11286475/tilt_uint8.mrc.sha512.zip)

**Review by dzenanz — APPROVED** [review]

**Comment by dzenanz** [comment]
Sorry, I merged these via rebase (without a merge commit). I now realize that will prevent same hashes from being used in release branch (#4023).

**Pull request merged**